java/cuvs-java/src/main/java/com/nvidia/cuvs/spi/CuVSProvider.java

```diff
@@ -52,9 +53,12 @@ default Path nativeLibraryPath() {
   /** Create a {@link Dataset.Builder} instance **/
   Dataset.Builder newDatasetBuilder(int size, int dimensions);
 
-  /** Create a {@link Dataset} backed by a on-heap array **/
+  /** Create a {@link Dataset} from an on-heap array **/
   Dataset newArrayDataset(float[][] vectors);
 
+  /** Create a {@link CagraGraph} from an on-heap array **/
+  CagraGraph newArrayGraph(int[][] graph);
```

Comment:
This is here mainly for test purposes; I do not anticipate a scenario where a cagra graph is built in Java, but it's useful to have it as a utility.